modules/sdk/src/test/scala/org/tessellation/sdk/infrastructure/snapshot/GlobalSnapshotStateChannelManagerSuite.scala

```diff
@@ -0,0 +1,196 @@
+    } yield expect.same((SortedMap(address -> NonEmptyList.one(stateChannelOutput.snapshotBinary)), Set.empty), result)
+
+  }
+
+  test("state channel should be discarded") { res =>
+    implicit val (kryo, sp) = res
+    val address = Address("DAG0y4eLqhhXUafeE3mgBstezPTnr8L3tZjAtMWB")
+
+    for {
+      stateChannelOutput <- mkStateChannelOutput(1, address, Some(Hash("unknown")))
+      snapshotInfo = mkGlobalSnapshotInfo(SortedMap.empty)
+      manager <- mkManager()
+      _ <- manager.accept(SnapshotOrdinal(1L), snapshotInfo, List(stateChannelOutput))
+      result <- manager.accept(SnapshotOrdinal(11L), snapshotInfo, List(stateChannelOutput))
+      expected = (SortedMap.empty[Address, NonEmptyList[Signed[StateChannelSnapshotBinary]]], Set.empty)
+    } yield expect.same(expected, result)
+
+  }
+
+  test("state channel with more signatures should be accepted and other should be discarded") { res =>
+    implicit val (kryo, sp) = res
+    val address = Address("DAG0y4eLqhhXUafeE3mgBstezPTnr8L3tZjAtMWB")
+
+    for {
+      stateChannelOutput1 <- mkStateChannelOutput(1, address, None)
+      stateChannelOutput2 <- mkStateChannelOutput(2, address, None)
+      snapshotInfo = mkGlobalSnapshotInfo(SortedMap.empty)
+      manager <- mkManager()
+      _ <- manager.accept(SnapshotOrdinal(1L), snapshotInfo, List(stateChannelOutput1))
+      _ <- manager.accept(SnapshotOrdinal(10L), snapshotInfo, List(stateChannelOutput1, stateChannelOutput2))
+      result <- manager.accept(SnapshotOrdinal(11L), snapshotInfo, List(stateChannelOutput1, stateChannelOutput2))
+      expected = (SortedMap(address -> NonEmptyList.one(stateChannelOutput2.snapshotBinary)), Set.empty)
+    } yield expect.same(expected, result)
+
+  }
+
+  test("state channel with more sent events should be accepted and other should be discarded") { res =>
+    implicit val (kryo, sp) = res
+    val address = Address("DAG0y4eLqhhXUafeE3mgBstezPTnr8L3tZjAtMWB")
+
+    for {
+      stateChannelOutput1 <- mkStateChannelOutput(1, address, None)
+      stateChannelOutput2 <- mkStateChannelOutput(1, address, None, stateChannelOutput1.snapshotBinary.value.some)
+      stateChannelOutput3 <- mkStateChannelOutput(1, address, None)
+      snapshotInfo = mkGlobalSnapshotInfo(SortedMap.empty)
+      manager <- mkManager()
+      _ <- manager.accept(SnapshotOrdinal(1L), snapshotInfo, List(stateChannelOutput1, stateChannelOutput2, stateChannelOutput3))
+      result <- manager.accept(SnapshotOrdinal(11L), snapshotInfo, List(stateChannelOutput1, stateChannelOutput2, stateChannelOutput3))
+      expected1 = (SortedMap(address -> NonEmptyList.one(stateChannelOutput2.snapshotBinary)), Set.empty)
+      expected2 = (SortedMap(address -> NonEmptyList.one(stateChannelOutput1.snapshotBinary)), Set.empty)
+    } yield expect.same(expected1, result) || expect.same(expected2, result)
+
+  }
+
+  test("state channels and proper following state channels should be accepted") { res =>
```

Comment:
maybe `state channels should form a chain`? or `... should be accepted in order`?

Reply:
Rephrased.